docs: Update Breadcrumbs stories to mdx (#1206)
Workday/canvas-kit: modules/preview-react/breadcrumbs/stories/Breadcrumbs.stories.mdx

<ExampleCodeBlock code={CollapsibleList} />

### Right-to-Left Example

Breadcrumbs has bidirectional support out of the box. That means outside of setting the content
direction in your application's Canvas theme, you don't need to do anything else to make it work.
All you need to supply is the translated text for items and ARIA labels.

<ExampleCodeBlock code={RTLCollapsibleList} />

## Components

### Breadcrumbs.Nav

#### Usage

`Breadcrumbs.Nav` a styled `<nav>` element. It has no intrinsic state or behaviors.

#### Props

<ArgsTable of={Breadcrumbs.Nav} />

> Note: This component also supports
> [all native HTMLElement props](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes).

---

### Breadcrumbs.List

#### Usage

`Breadcrumbs.List` component is a styled, unordered list (`<ul>`) with no additional behaviors. Use
this when you're not concerned about collapsing the list of breadcrumb items into a dropdown menu at
a specified width.

#### Props

`Breadcrumbs.List` has no component-specific props. However, it supports
[all native `<ul>` element props](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#Attributes).

> ---

Author: @jamesfan

How do you feel about dividers between components and subcomponents? I didn't see any on `Card`, but I added them here to break up the content a bit. I don't have a strong preference, but we should probably pick one and stick to it.

import {Meta} from '@storybook/addon-docs/blocks';
import {Breadcrumbs} from '@workday/canvas-kit-preview-react/breadcrumbs';
import {Specifications} from '@workday/canvas-kit-docs';

import {StandardList} from './examples/StandardList';
import {CollapsibleList} from './examples/CollapsibleList';
import {RTLCollapsibleList} from './examples/RTL';
import {CurrentItemTruncation} from './examples/CurrentItemTruncation';
import {LinkTruncation} from './examples/LinkTruncation';

<Meta title="Preview/Breadcrumbs/React" component={Breadcrumbs} />

# Canvas Kit Breadcrumbs

Breadcrumbs allow users to keep track and maintain awareness of their location as they navigate
through pages, folders, files, etc.

[> Workday Design Reference](https://design.workday.com/components/navigation/breadcrumbs)

## Installation

```sh
yarn add @workday/canvas-kit-preview-react
```

## Usage

Breadcrumbs is a composable component that provides some flexibility in it's structure. There are
two main variations: standard list and collapsible list. We'll discuss the standard list first. All

> subcomponents are described in more detail in the [Components](#components) section below.

Author: Yeah, good call out. I initially wrote this paragraph when we were first introducing the idea of composable components. On a second read, I think it feels too close to compound components (which it isn't), and it could be easily confused. I'll remove it. 👍🏻

> ### Standard List Example

Author: I'm not too particular about sorting stories in the Storybook sidenav, but I'm good with following the convention for the docs. 👍🏻

`Breadcrumbs.List` is a styled unordered list `<ul>` with no additional behaviors. Use this when you
are not concerned about collapsing the list of breadcrumb items into a dropdown menu at a specified
width.

Accessibility is built into Breadcrumbs components in a way that allows you to create an inclusive
experience with little additional configuration. The only ARIA label you'll need to add is the
`aria-label` for `Breadcrumbs.Nav` as seen in the example. And all links require `href`s to be
properly identified. All other accessible attributes are baked into the components.

<ExampleCodeBlock code={StandardList} />

### Collapsible List Example

`Breadcrumbs.CollapsibleList` is built on top of `Breadcrumbs.List`. It provides additional
functionality to collapse list items into a dropdown menu at a specified width.

> Accessibility is built into Breadcrumbs components in a way that allows you to create an inclusive
> experience with little additional configuration. The only ARIA labels you'll need to add are the
> `aria-label` for `Breadcrumbs.Nav` and the `buttonAriaLabel` for `Breadcrumbs.CollapsibleList`. And
> all links require `href`s to be properly identified. All other accessible attributes are baked into
> the components.

Reviewer: Nit: could shorten this paragraph given that it's almost identical word-for-word to the paragraph in the `Standard List Example` section. Something like this:

```suggestion
In addition to adding `aria-label` to `Breadcrumbs.Nav` and identifying all links with an `href`,
you'll need to add `buttonAriaLabel` to `Breadcrumbs.CollapsibleList`.
```

<ExampleCodeBlock code={CollapsibleList} />

### Right-to-Left Example

Breadcrumbs has bidrectional support out of the box. That means outside of setting the content
direction in your application's Canvas theme, you don't need to do anything else to make it work.
All you need to supply is the translated text for items and ARIA labels.

<ExampleCodeBlock code={RTLCollapsibleList} />

## Components

### Breadcrumbs.Nav

#### Usage

`Breadcrumbs.Nav` a styled `<nav>` element. It has no intrinsic state or behaviors.

#### Props

<ArgsTable of={Breadcrumbs.Nav} />

> Note: This component also supports

> > [all native HTMLElement props](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes).

Author: Yeah, I like that better than the indented note. 👍🏻

the list will never be collapsed.

##### Handling Custom Dropdown Button Icons

By default the icon for the dropdown button is `relatedActionsIcon`. However, you can set this to
any [Canvas System Icon](https://design.workday.com/tokens/assets/icons/system-icons) with the
`buttonIcon` prop. Please consult the Canvas team if you decide to do this, as we'd like to keep
breadcrumb navigation as consistent as possible.

#### Props

<ArgsTable of={Breadcrumbs.CollapsibleList} />

> Note: This component also supports
> [all native `<ul>` element props](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#Attributes).

---

### Breadcrumbs.ListItem

#### Usage

`Breadcrumbs.ListItem` is a styled `<li>` element that contains a `SystemIcon` separator.

#### Props

`Breadcrumbs.ListItem` has no component-specific props. However, it supports
[all native `<li>` element props](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li#Attributes).

---

### Breadcrumbs.Link

#### Usage

`Breadcrumbs.Link` is a styled `<a>` element that also renders a tooltip if the text is truncated.
The built-in truncation + tooltip functionality provides an easy-to-use, accessible feature for

> managing the length of link text. By default, the `maxWidth` is set to `350px`, however you can
> adjust this as needed. Note that tooltips will only render when text is truncated. Here's an example
> with the `maxWidth` set to `150px`.

Reviewer: ```suggestion
managing the length of link text. The `maxWidth` is set to `350px` by default and can be adjusted as
needed. Note that tooltips will only render when text is truncated. Here's an example with the
`maxWidth` set to `150px`.
```

`history` package.

```tsx
import * as React from 'react';
// using the history package
import {createBrowserHistory} from 'history';
import {Breadcrumbs} from '@workday/canvas-kit-preview-react/breadcrumbs';

// this is likely being done at the root of your application and not inside this component.
const history = createBrowserHistory();

const handleClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
  // handle any click side-effects here
  console.log('click event', event);
}

const handleRedirect = (href: string) => {
  // redirect using the internal SPA router
  history.push(href);
}

return (
  <Breadcrumbs.Link
    href="/account"
    onAction={handleRedirect}
    onClick={handleClick}
  >
    Account
  </Breadcrumbs.Link>;
)
```

#### Props

<ArgsTable of={Breadcrumbs.Link} />

> Note: This component also supports
> [all native `<a>` element props](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Attributes).

---

### Breadcrumbs.CurrentItem

#### Usage

`Breadcrumbs.CurrentItem` is a styled `<li>` element also renders a tooltip if the text is
truncated. The built-in truncation + tooltip functionality provides an easy-to-use, accessible

> feature for managing the length of link text. By default, the `maxWidth` is set to `350px`, however
> you can adjust this as needed. Normally, this is a non-focusable element. But when truncated, it
> sets the `tabIndex` to `0` to enable the tooltip to appear on keyboard focus. Note that tooltips
> will only render when text is truncated. Here's an example with the `maxWidth` set to `100px`.

Reviewer: ```suggestion
feature for managing the length of link text. The `maxWidth` is set to `350px` by default and can be
adjusted as needed. Normally, this is a non-focusable element. But when truncated, it sets the
`tabIndex` to `0` to enable the tooltip to appear on keyboard focus. Note that tooltips will only
render when text is truncated. Here's an example with the `maxWidth` set to `100px`.
```
